gca/core.py

```diff
@@ -569,10 +571,10 @@ def is_authenticated(self):
         return self.__is_authenticated
 
     def authenticate(self):
-        purl = urlparse(self.url)
+        purl = parse.urlparse(self.url)
         hostname = purl.hostname
         user, password = self.__auth.get_credentials(hostname)
-        params = urllib.urlencode({'identifier': user, 'password': password})
+        params = urllib.parse.urlencode({'identifier': user, 'password': password})
         url_opener = self.__url_opener
         resp = url_opener.open(self.url + "/authenticate/credentials", params)
```

Comment:
Auth params need to be byte encoded for the URL
```suggestion
         resp = url_opener.open(self.url + "/authenticate/credentials", params.encode("utf-8"))
```